translated_mock_driver_testing/CMakeLists.txt

```diff
@@ -0,0 +1,139 @@
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  cav_msgs
+  cav_srvs
+  roscpp
+)
+
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+#  LIBRARIES ui_integration
+   CATKIN_DEPENDS cav_msgs cav_srvs roscpp 
+)
+#  DEPENDS system_lib
+
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+)
+
+
+## Declare a C++ executable
+## With catkin_make all packages are built within a single CMake context
+## The recommended prefix ensures that target names across packages don't collide
+file(GLOB_RECURSE headers */*.hpp */*.h)
+add_executable(${PROJECT_NAME}_node
+  src/main.cpp
+  src/MockLightBarDriver.cpp
+)
+
+add_library(${PROJECT_NAME}_lib
+                src/main.cpp
+                src/MockLightBarDriver.cpp
+                
+)
+
+
+
+## Add cmake target dependencies of the executable
+add_dependencies(${PROJECT_NAME}_node 
+                ${catkin_EXPORTED_TARGETS}
+)
+
+add_dependencies(${PROJECT_NAME}_lib
+                ${catkin_EXPORTED_TARGETS}
+) 
+
+
+## Specify libraries to link a library or executable target against
+target_link_libraries(${PROJECT_NAME}_node
+  ${catkin_LIBRARIES}
+)
+
+target_link_libraries(${PROJECT_NAME}_lib
+  ${catkin_LIBRARIES}
+)
+
+
+#############
+## Install ##
+#############
+
+## Mark cpp header files for installation
+install(DIRECTORY include/${PROJECT_NAME}
+DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+FILES_MATCHING PATTERN "*.hpp"
+PATTERN ".svn" EXCLUDE
+)
+
+## Mark executables and/or libraries for installation
+install(TARGETS ${PROJECT_NAME}_node
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
+
+
+## Mark other files for installation (e.g. launch and bag files, etc.)
+install(DIRECTORY include/${PROJECT_NAME}/
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+
+install(DIRECTORY launch/
+        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/launch
+        FILES_MATCHING PATTERN "*.launch"
+        PATTERN ".svn" EXCLUDE)
+
+#############
+## Testing ##
+#############
+
+## Add gtest based cpp test target and link libraries
+catkin_add_gtest(${PROJECT_NAME}-test 
```

Comment:
As mentioned below, please see guidance and plan_delegator for how to include rostest. Because currently, the Circle CI is failing for your unit test because it requires active roscore to run so you need rostest package not this type of test. I have not tested locally if it will work, but let's try adjusting CMakelists and .test file according to these examples, and let me know if you need any further assistance.